sql/src/main/java/io/crate/executor/transport/AlterTableOperation.java

```diff
@@ -107,7 +108,64 @@ public AlterTableOperation(ClusterService clusterService,
         return result;
     }
 
-    private class ResultSetReceiver implements ResultReceiver {
+    public CompletableFuture<Long> executeAlterTableOpenClose(final AlterTableOpenCloseAnalyzedStatement analysis) {
+        List<CompletableFuture<Long>> results = new ArrayList<>(2);
+        DocTableInfo table = analysis.table();
+
+
+
+        if (table.isPartitioned()) {
+            Optional<PartitionName> partitionName = analysis.partitionName();
+            if (partitionName.isPresent()) {
+                String idx = partitionName.get().asIndexName();
+
+                if (analysis.openTable()) {
+                    results.add(openTable(idx));
+                } else {
+                    results.add(closeTable(idx));
+                }
+            } else {
+                LinkedHashMap<String, Object> metaMap = new LinkedHashMap<>();
+                if (!analysis.excludePartitions() && analysis.table().partitions().size() > 0) {
+                    String[] indices = Stream.of(table.concreteIndices()).toArray(String[]::new);
+                    if (analysis.openTable()) {
+                        results.add(openTable(indices));
+                    } else {
+                        results.add(closeTable(indices));
+                        metaMap.put("_meta", new LinkedHashMap(){{put("closed", true);}});
+                        results.add(updateTemplate(metaMap, null, table.ident()));
+                    }
+                }
+            }
+        } else {
+            if (analysis.openTable()) {
+                results.add(openTable(table.ident().indexName()));
+            } else {
+                results.add(closeTable(table.ident().indexName()));
+            }
+        }
+
+        final CompletableFuture<Long> result = new CompletableFuture<>();
+        applyMultiFutureCallback(result, results);
+        return result;
```

Comment:
this all above is not needed at all, you'll only have 1 future to return. so directly return the future of close/openTable()